guides/keras_cv/cut_mix_mix_up_and_rand_augment.py

```diff
@@ -0,0 +1,402 @@
+
+Perhaps you want to exclude an augmentation from `RandAugment`, or perhaps you want to
+include the `GridMask()` as an option alongside the default `RandAugment` augmentations.
+
+KerasCV allows you to construct production grade custom data augmentation pipelines using
+the `keras_cv.layers.RandomAugmentationPipeline` layer. This class operates similarly to
+`RandAugment`; selecting a random layer to apply to each image `augmentations_per_image`
+times. `RandAugment` can be thought of as a specific case of
+`RandomAugmentationPipeline`. In fact, our `RandAugment` implementation inherits from
+`RandomAugmentationPipeline` internally.
+
+In this example, we will create a custom `RandomAugmentationPipeline` by removing
+`RandomRotation` layers from the standard `RandAugment` policy, and substitutex a
+`GridMask` layer in its place.
+"""
+
+"""
+As a first step, let's use the helper method `RandAugment.get_standard_policy()` to
+create a base pipeline.
+"""
+
+layers = keras_cv.layers.RandAugment.get_standard_policy(
+    value_range=(0, 255), magnitude=0.75, magnitude_stddev=0.3
+)
+
+"""
+First, let's filter out `RandomRotation` layers
+"""
+
+layers = [
+    layer for layer in layers if not isinstance(layer, keras_cv.layers.RandomRotation)
+]
+
+"""
+Next, let's add `GridMask` to our layers:
+"""
+
+layers = layers + [keras_cv.layers.GridMask()]
+
+"""
+Finally, we can put together our pipeline
+"""
+
+pipeline = keras_cv.layers.RandomAugmentationPipeline(
+    layers=layers, augmentations_per_image=3
+)
+
+"""
+Let's check out the results!
+"""
+
+
+@tf.function
+def apply_pipeline(inputs):
+    inputs["images"] = pipeline(inputs["images"])
+    return inputs
+
+
+train_dataset = load_dataset().map(apply_pipeline, num_parallel_calls=AUTOTUNE)
+visualize_dataset(train_dataset, title="After custom pipeline")
+
+"""
+Awesome! As you can see, no images were randomly rotated. You can customize the
+pipeline however you like:
+"""
+
+pipeline = keras_cv.layers.RandomAugmentationPipeline(
+    layers=[keras_cv.layers.GridMask(), keras_cv.layers.Grayscale(output_channels=3)],
+    augmentations_per_image=1,
+)
+
+"""
+This pipeline will either apply `GrayScale` or GridMask:
+"""
+
+
+@tf.function
+def apply_pipeline(inputs):
+    inputs["images"] = pipeline(inputs["images"])
+    return inputs
+
+
+train_dataset = load_dataset().map(apply_pipeline, num_parallel_calls=AUTOTUNE)
+visualize_dataset(train_dataset, title="After custom pipeline")
+
+"""
+Looks great! You can use `RandomAugmentationPipeline` however you want.
+"""
+
+"""
+## Training a CNN
+
+As a final exercise, let's take some of these layers for a spin. In this section, we
+will use `CutMix`, `MixUp`, and `RandAugment` to train a state of the art `ResNet50`
+image classifier on the Oxford flowers dataset.
+
+"""
+
+
+def preprocess_for_model(inputs):
+    images, labels = inputs["images"], inputs["labels"]
+    images = tf.cast(images, tf.float32) / 255.0
```

Comment:
Perhaps we should encourage users to make rescaling part of the model itself (so that it can handle raw images without end users having to worry about which particular normalization / rescaling scheme it requires).

Reply:
Fixed